### PR TITLE
ocsp: don't free ocsp request if saved in ssl->ctx->certOcspRequest

### PR DESCRIPTION
**Pull request by rizlik**

### CODE REVIEW

**Comment by rizlik** [comment]
retest this please

**Review by dgarske — APPROVED** [review]

File: src/internal.c [hunk 1]
```diff
@@ -23310,8 +23310,10 @@ int SendFinished(WOLFSSL* ssl)
  * Returns 0 on success
  */
 static int CreateOcspRequest(WOLFSSL* ssl, OcspRequest* request,
-                             DecodedCert* cert, byte* certData, word32 length)
+                             DecodedCert* cert, byte* certData, word32 length,
+                             byte *takeOwnership)
```
Comment (SparkiDev) [inline src/internal.c:23314]: takeOwnership => ctxHasRequest

Why not update pointer in lock protected code?

File: src/internal.c [hunk 2]
```diff
@@ -23397,7 +23404,7 @@ int CreateOcspResponse(WOLFSSL* ssl, OcspRequest** ocspRequest,
         createdRequest = 1;
         if (ret == 0) {
             ret = CreateOcspRequest(ssl, request, cert, der->buffer,
-                                                                   der->length);
+                      der->length, &ctxOwnsRequest);
```
Comment (SparkiDev) [inline src/internal.c:23407]: Can reuse createdRequest as freeRequest and thenCreateOcspRequest sets the value to 0 when ctx takes ownership.

**Review by SparkiDev — CHANGES_REQUESTED** [review]

File: src/internal.c (hunk 2 above)
Comment (rizlik) [reply src/internal.c:23407]: This saves a variable but makes the code slightly more cumbersome to read. Is it worthwhile?
(CreateOcspRequest should use inverted logic for the variable and can't set freeRequest to zero when  not taking ownership of the variable)

**Review by SparkiDev — APPROVED** [review]

**Pull request merged**